shimmingtoolbox/cli/b0shim.py

```diff
@@ -1018,32 +1033,58 @@ def _plot_coefs(coil, slices, static_coefs, path_output, coil_number, rt_coefs=N
         temp_max = np.array(static_coefs).max()
         if max_y is None or max_y < temp_max:
             max_y = np.array(static_coefs).max()
+    for nb_slices, i_shim in enumerate(shimmed_slice):
+        i_slice = slices[i_shim]
+        _add_sub_figure(i_shim, fig, number_slices_shimmed, nb_slices+1, rt_coefs, pres_probe_min,
+                        pres_probe_max, bounds, min_y, max_y,
+                        units, static_coefs, i_slice)
+    # Print a subplot for all the non shimmed slices
+
+    if unused_slice:
+        nb_slices = nb_slices + 1
+        _add_sub_figure(last_i, fig, number_slices_shimmed, nb_slices+1, rt_coefs, pres_probe_min,
+                    pres_probe_max, bounds, min_y, max_y,
+                    units, static_coefs, list_slice_wo_shimmed)
+    # Save the figure
+    fname_figure = os.path.join(path_output, f"fig_currents_per_slice_group_coil{coil_number}_{coil.name}.png")
+    fig.savefig(fname_figure, bbox_inches='tight')
+    logger.debug(f"Saved figure: {fname_figure}")
 
-    # Create a plot for each shim group
-    for i_shim in range(n_shims):
-        ax = fig.add_subplot(n_shims + 1, 1, i_shim + 1)
-        n_channels = static_coefs.shape[1]
-
-        # Add realtime component as an errorbar
-        if rt_coefs is not None:
-            rt_coef_ishim = rt_coefs[i_shim]
-            riro = [rt_coef_ishim * -pres_probe_min, rt_coef_ishim * pres_probe_max]
-            ax.errorbar(range(n_channels), static_coefs[i_shim], yerr=riro, fmt='o', elinewidth=4, capsize=6,
-                        label='static-riro')
-        # Add static component
-        else:
-            ax.scatter(range(n_channels), static_coefs[i_shim], marker='o', label='static')
 
-        # Draw a black line at y=0
-        ax.hlines(0, 0, 1, transform=ax.get_yaxis_transform(), colors='k')
+def _add_sub_figure(i_shim, fig, n_shims, axis, rt_coefs, pres_probe_min, pres_probe_max, bounds, min_y, max_y,
+                    units, static_coefs, i_slice):
```

Comment:
n_shims seems to be called with the same number whether there are unused slices are not.

Reply:
Yes, You would prefer to redeclare nshims, everytime that we enter the function? I thaught it would be better to do it in plot_coefs because we get this variable with slice in plot_coefs, and so it is better to only calculate it once for me.

Reply:
I meant whether there are unused slices or not, n_shims is still the same number which should not occur since n)shims describes the number of subplots. You are not seeing an error because you added 1 and are not using the last subplot if there are no unshimmed slice. I fixed it.